app/controllers/home_controller.rb

```diff
@@ -20,30 +18,29 @@ def home
       @samples = sort_by(Sample.includes(:pipeline_runs, :pipeline_outputs).paginate(page: params[:page]), sort)
       @samples_count = Sample.all.size
     end
+    @final_result = samples_info(@samples) ? samples_info(@samples)[:final_result] : nil
```

Comment:
instead of calling samples_info 4 times, I think you should just call int one time. 

basically do this

samples_info_output = samples_info(@samples) || {}
@final_results = samples_info_output[:final_result]
@pipeline_run_info = samples_info_output[:pipeline_run_info]

Reply:
Great, thanks!